src/jit/flowgraph.cpp

```diff
@@ -9769,49 +9766,40 @@ void                Compiler::fgCompactBlocks(BasicBlock* block, BasicBlock* bNe
     // control-flow choice was constant-folded away.  So probably phi's need to go away,
     // as well, in favor of one of the incoming branches.  Or at least be modified.
 
-    assert((block->IsLIR() == bNext->IsLIR()) || (block->IsLIR() && bNext->isEmpty()) || (block->isEmpty() && bNext->IsLIR()));
+    assert(block->IsLIR() == bNext->IsLIR());
     if (block->IsLIR() || bNext->IsLIR())
```

Comment:
The "if" below could be simplified based on this assert (only need to check one)